packages/components/addon/components/hds/dropdown/list-item/copy-item.js

```diff
@@ -37,12 +37,19 @@ export default class HdsDropdownListItemCopyItemComponent extends Component {
   }
 
   @action
-  copyCode() {
-    navigator.clipboard.writeText(this.args.text);
-    // this if statement resolves to [object Promise] so maybe some improvements
-    // could be made here
-    if (navigator.clipboard.readText()) {
+  async copyCode() {
+    // https://developer.mozilla.org/en-US/docs/Web/API/Clipboard
+    await navigator.clipboard.writeText(this.args.text);
+    const result = await navigator.clipboard.readText();
+
+    if (result === this.args.text) {
```

Comment:
[nitpick/improvement] since we have a "getter" function for the value `this.text`, and potentially thiw function could be manipulating the value in the future, it would be more resilient if here we use `this.text` (and at line 41 too) instead of the argument passed.

Reply:
there's a reason to use this.args.text but I need to look it up. I'll find the text so we can all have the same knowledge and will update this PR if I am not correct.